chapter9.adoc

```diff
@@ -0,0 +1,372 @@
+when MXLEN=64 and <<MRMCFG32>> when MXLEN=32. The CSR number is TBA.
+
+The `RCID` and `MCID` accompany each request made by the hart, with effective
+privilege mode equal to M, to shared resource controllers . The `RCID` is used to
+determine the resource allocations (e.g., cache occupancy limits, memory
+bandwidth limits, etc.) to enforce. The `MCID` is used to identify a counter to
+monitor resource usage.
+
+[NOTE]
+====
+A suggested reset value of 0 for the `RCID` field within the `mrmcfg` register
+aligns with the default operational behavior of resource controllers, which is
+to associate all capacity and resources with `RCID=0`. This setup ensures a
+seamless initial state that conforms to expected default resource allocation
+policies. The reset value for the `MCID` field, being potentially
+implementation-defined, does not directly impact core functionality and may
+vary based on specific system requirements.
+====
+
+When Smqosid is implemented, the configurations in the `srmcfg` register
+introduced by Ssqosid extension only apply to requests made with effective
+privilege modes less than M.
+
+[[MRMCFG64]]
+.Machine Resource Management Configuration (`mrmcfg`) register for MXLEN=64
+
+[wavedrom, , ]
+....
+{reg: [
+  {bits: 12, name: 'RCID'},
+  {bits:  4, name: 'WPRI'},
+  {bits: 12, name: 'MCID'},
+  {bits: 36, name: 'WPRI'},
+], config:{lanes: 2, hspace:1024}}
+....
+
+[[MRMCFG32]]
+.Machine Resource Management Configuration (`mrmcfg`) register for MXLEN=32
+
+[wavedrom, , ]
+....
+{reg: [
+  {bits: 12, name: 'RCID'},
+  {bits:  4, name: 'WPRI'},
+  {bits: 12, name: 'MCID'},
+  {bits:  4, name: 'WPRI'},
+], config:{lanes: 1, hspace:1024}}
+....
+
+The Smqosid extension enables the scenario where two or more security domains
+share a common set of capacity and bandwidth controllers. In this setup, access
+to the QRI of these controllers is restricted to RDSM, which then provides
+mediated access for these security domains to configure capacity or bandwidth
+allocation and read the corresponding monitoring counters. A common `QRID` is
+associated with requests from these domains, and the RDSM allocates sets of
+`RCID` and `MCID` for each domain's use. Supervisor domains are then restricted
+to selecting a `RCID` and `MCID` in `srmcfg` from the allocated range. Although
+the RDSM could limit access to `srmcfg` to prevent direct domain configuration,
+enabling domains to program `srmcfg` directly is preferred for performance
+reasons. To facilitate this, the Smqosid extension introduces `SRL`
+(supervisor-RCID-length) and `SML` (supervisor-MCID-length) fields in the
+`msdcfg` register to configure the usable range of `RCID` and `MCID`. These
+fields are WARL, with legal values from 0 to 12. Furthermore, a
+supervisor-srmcfg-mode (`SSM`) field is introduced, with legal values of 0 and
+1. These settings, along with `SRL` and `SML`, determine the range of `RCID` and
+`MCID` values an SD can program into the `srmcfg` register.
```

Comment:
clarify requirement is for these ranges to be disjoint

Reply:
done - in new PR